### ISSUE TITLE
OPUS causing Video frame getting stuck when seeking

### ISSUE THREAD

**Issue by XxAcielxX** [anxdpanic/plugin.video.youtube#537]
### Context

Please provide any relevant information about your setup

* Add-on Version: 7.0.2.2
* Kodi Version: 20.2.0
* Kodi GUI Language: English
* Operating System: LibreELEC 11.0.3
* Operating System Language: English

------

### Expected Behavior

Seeking video back or forward should keep the video & audio in sync. Video should not get stuck on first frame after seeing for few seconds.

------

### Current Behavior

When seeking the video back or forward, video gets stuck on first frame after seeking, while the audio keeps playing. Then after few seconds, video starts to play in sync.

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Play any video
2. Seek back or forward, like 10secs 

------

### Additional Information

I have found the culprit here, when the video is played using OPUS, this issue occurs. If AAC is selected, it doesn't happen at all. 

I have to unselect OPUS in Stream Selection. Though my SBC (NanoPC-T4) is able to play OPUS fine, so this must be a bug. 

------

**Comment by MoojMidge** [comment]
> I have found the culprit here, when the video is played using OPUS, this issue occurs. If AAC is selected, it doesn't happen at all.
> 
> I have to unselect OPUS in Stream Selection. Though my SBC (NanoPC-T4) is able to play OPUS fine, so this must be a bug.

The problem is not exclusive to playing Opus audio, but it does seem to be more noticeable.

Unfortunately not a bug with the plugin, but try with a LE12 build and latest available version of InputStream.Adaptive 21, should be a lot better.

**Comment by XxAcielxX** [comment]
I don't feel like using non-stable LibreELEC build just for testing here, as downgrading will be impossible then.

I'll keep using AAC, and see if any changes are made from upstream regarding this issue on Nexus. I'll keep you updated.

**Comment by MoojMidge** [comment]
> I don't feel like using non-stable LibreELEC build just for testing here, as downgrading will be impossible then.

Fair enough. It is possible to create a backup, upgrade using a clean install for testing, and then re-install an older version with your backup. Or if you have different installation media that can be swapped out, it is even easier. But it can be a bit of a hassle.

Only some of the related changes to AudioEngine were backported to Nexus, so you probably won't see any major improvement until you install LE12.

Having said that, I think I spoke too soon regarding it being "a lot better". It does seem improved, but still noticably worse when playing Opus audio.

**Comment by MoojMidge** [comment]
Going to disable Opus audio by default in `Stream features` in the next release as a temporary workaround.

**Comment by Kellett** [comment]
For anyone else having this problem but struggling to find the settings...

1. YouTube Addon settings
2. MPEG-DASH from the side menu
3. Click "Stream Features" 
4. Untick "Enable Opus Audio" from the popup

After doing this I had no more problems with skipping in videos. All videos I tested worked fine.

**Comment by onlytanmoy** [comment]
@Kellett working like a charm now, many thanks mate, happy holidays :)

**Comment by XxAcielxX** [comment]
* Add-on Version: 7.0.7+beta.3
* Kodi Version: 21.0.0
* Kodi GUI Language: English
* Operating System: LibreELEC 12.0.0
* Operating System Language: English

Retested with Kodi 21 release, and the issue is still present.

**Comment by MoojMidge** [comment]
Yes, I reverted the change to default settings in the plugin, but the changes in Kodi that I thought would resolve this don't directly address this specific issue.

**Comment by Miteirao** [comment]
2026, still no way to play YouTube videos VP9+opus?
